reconcile/terraform_aws_route53.py

```diff
@@ -116,6 +119,50 @@ def build_desired_state(zones):
                 logging.debug(msg)
                 record["records"] = record_values
 
+            # Process '_target_namespace_zone'
+            target_namespace_zone = record.pop("_target_namespace_zone", None)
+            if target_namespace_zone:
+                tf_resources = target_namespace_zone["namespace"]["terraformResources"]
+                tf_zone_name = target_namespace_zone["name"]
+                tf_zone_resources = [
+                    tfr
+                    for tfr in tf_resources
+                    if tfr["provider"] == "route53-zone" and tfr["name"] == tf_zone_name
+                ]
+                if not tf_zone_resources:
+                    logging.error(
+                        f"{zone_name}: field `_target_namespace_zone` found "
+                        f"for record {record_name}, but target zone not found: "
+                        f"{tf_zone_name}"
+                    )
+                    sys.exit(ExitCodes.ERROR)
+                tf_zone_resource = tf_zone_resources[0]
+                tf_zone_account_name = tf_zone_resource["account"]
+                zone_account = [
+                    a for a in all_accounts if a["name"] == tf_zone_account_name
+                ][0]
+                awsapi = AWSApi(1, [zone_account], settings=settings, init_users=False)
+                tf_zone_region = (
+                    tf_zone_resource.get("region")
+                    or zone_account["resourcesDefaultRegion"]
+                )
+                tf_zone_ns_records = awsapi.get_route53_zone_ns_records(
+                    tf_zone_account_name, tf_zone_name, tf_zone_region
+                )
+                if not tf_zone_ns_records:
+                    logging.warning(
+                        f"{zone_name}: field `_target_namespace_zone` found "
+                        f"for record {record_name}, but target zone not found (yet): "
+                        f"{tf_zone_name}"
+                    )
+                    continue
+                logging.debug(
+                    f"{zone_name}: field `_target_namespace_zone` found "
+                    f"for record {record_name}, Values are: "
+                    f"{tf_zone_ns_records}"
+                )
+                record["records"] = tf_zone_ns_records
```

Comment:
Similarly, the output of the above extracted function could then be used to query and aggregate the real-world zone records.

```
def _query_zone_info(zone_account: ZoneAccount) -> list[tf_zone_ns_records?]:
    pass
```